scan/lib/scan/runner.rb

```diff
@@ -39,11 +39,20 @@ def test_app
         }
       ]
       exit_status = 0
+
+      app_identifier = Scan.config[:app_identifier]
+      app_identifier ||= UI.input("App Identifier: ")
+
+      Scan.devices.each do |device|
+        FastlaneCore::Simulator.uninstall_app(app_identifier, device.name, device.udid) if Scan.config[:reinstall_app]
```

Comment:
This `if Scan.config[:reinstall_app]` should go before the `Scan.devices.each` so we don't have to iterate over devices if we don't have to 😊